tests/unit/sync/elementTransformers.test.ts

```diff
@@ -0,0 +1,413 @@
+      },
+      default: {
+        global: {
+          value: [
+            {
+              codename: taxonomyGroups[0].terms[0].codename,
+            },
+            {
+              codename: taxonomyGroups[0].terms[0].terms[0].codename,
+            },
+          ],
+        },
+      },
+    };
+
+    const transformedElement = transformTaxonomyElement(element, taxonomyGroups);
+
+    expect(transformedElement).toEqual(expectedOutput);
+  });
+
+  it("transformLinkedItemsElement correctly transforms element", () => {
+    const element: ContentTypeElements.ILinkedItemsElement = {
+      ...deafultObject,
+      type: "modular_content",
+      default: {
+        global: { value: [{ id: items[0].id }] },
+      },
+      allowed_content_types: [{ id: contentTypes[0].id }],
+    };
+
+    const expectedOutput = {
+      ...element,
+      id: undefined,
+      external_id: element.id,
+      default: {
+        global: {
+          value: [
+            {
+              codename: items[0].codename,
+              external_id: items[0].id,
+            },
+          ],
+        },
+      },
+      allowed_content_types: [{ codename: contentTypes[0].codename }],
+    };
+
+    const transformedElement = transformLinkedItemsElement(element, contentTypes, items, logOptions);
+
+    expect(transformedElement).toEqual(expectedOutput);
+  });
+
+  it("transformLinkedItemsElement correctly transforms element with missing item and types", () => {
+    const element: ContentTypeElements.ILinkedItemsElement = {
+      ...deafultObject,
+      type: "modular_content",
+      default: {
+        global: { value: [{ id: items[0].id }] },
+      },
+      allowed_content_types: [{ id: contentTypes[0].id }],
+    };
+
+    const expectedOutput = {
+      ...element,
+      id: undefined,
+      external_id: element.id,
+      default: undefined,
+      allowed_content_types: undefined,
+    };
+
+    const transformedElement = transformLinkedItemsElement(element, [], [], logOptions);
+
+    expect(transformedElement).toEqual(expectedOutput);
+  });
+
+  it("transformGuidelinesElement correctly transforms element", () => {
+    const guidelines = `<p>Item links: <a data-item-id="${
+      items[0].id
+    }">Item Link 1</a></p>\n<p>Asset Link: <a data-asset-id="${
+      assets[0].id
+    }">Asset Link 1</a></p>\n<figure data-asset-id="${assets[0].id}"><img src="#" data-asset-id="${
+      assets[0].id
+    }"></figure>`;
+    const element: ContentTypeElements.IGuidelinesElement = {
+      ...deafultObject,
+      type: "guidelines",
+      guidelines: guidelines,
+    };
+
+    const expectedOutput = {
+      ...element,
+      id: undefined,
+      external_id: element.id,
+      guidelines: `<p>Item links: <a data-item-codename="${items[0].codename}" data-item-external-id="${
+        items[0].id
+      }">Item Link 1</a></p>\n<p>Asset Link: <a data-asset-codename="${assets[0].codename}" data-asset-external-id="${
```

Comment:
You can use a real new line instead of the `\n`. White spaces have no effect on the rich-text parsing and it would make the code here more pleasant to eyes in my opinion.

Reply:
You are right. I forgot that we only use regexes not the rich-text parser to parse the string so the result is susceptible to white-space changes. Lets leave it as is for now then.